srml/contract/src/gas.rs

```diff
@@ -204,14 +205,14 @@ pub fn buy_gas<T: Trait>(
 ) -> Result<(GasMeter<T>, NegativeImbalanceOf<T>), &'static str> {
 	// Check if the specified amount of gas is available in the current block.
 	// This cannot underflow since `gas_spent` is never greater than `block_gas_limit`.
-	let gas_available = <Module<T>>::block_gas_limit() - <Module<T>>::gas_spent();
+	let gas_available = T::BlockGasLimit::get() - <Module<T>>::gas_spent();
 	if gas_limit > gas_available {
 		// gas limit reached, revert the transaction and retry again in the future
 		return Err(BLOCK_FULL);
 	}
 
 	// Buy the specified amount of gas.
-	let gas_price = <Module<T>>::gas_price();
+	let gas_price = T::GasPrice::get();
```

Comment:
I am not sure about motivation behind this, can you elaborate?

Reply:
Ok, I see. I don't think that gas price can be treated as a rarely changed parameter.

Reply:
The gas price isn't written to in the contracts module. So until now the only way to set the gas price was by configuring it in the genesis block.